### PR TITLE
Add persistent statements summary

### PR DESCRIPTION
**Pull request by mornyx**
### What is changed, added or deleted? (Required)

Add documentation for a new experimental feature: Persistent Statements Summary.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/12069
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by zhongzc — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]

File: statement-summary-tables.md [hunk 1]
```diff
@@ -177,6 +177,33 @@ Statement summary tables 现在还存在以下限制：
...
+[instance]
+tidb_stmt_summary_enable_persistent = true
+# 以下配置为默认值，可根据需求调整。
+# tidb_stmt_summary_filename = "tidb-statements.log"
+# tidb_stmt_summary_file_max_days = 3
+# tidb_stmt_summary_file_max_size = 64 # MB
+# tidb_stmt_summary_file_max_backups = 0
+```
+
+开启 statements summary 持久化后，内存中只维护当前的实时数据，不再维护历史数据，历史数据生成后直接被写入磁盘文件（写入周期参考 [参数配置](#参数配置) 一节所描述的 `tidb_stmt_summary_refresh_interval`）。后续针对 `statements_summary_history` 或 `cluster_statements_summary_history` 表的查询将结合内存和磁盘两处数据返回结果。
```
Comment (Oreoxmt) [inline statement-summary-tables.md:200]:
```suggestion
开启 statements summary 持久化后，内存中只维护当前的实时数据，不再维护历史数据。历史数据生成后直接被写入磁盘文件，写入周期参考[参数配置](#参数配置)一节所描述的 `tidb_stmt_summary_refresh_interval`。后续针对 `statements_summary_history` 或 `cluster_statements_summary_history` 表的查询将结合内存和磁盘两处数据返回结果。
```

File: statement-summary-tables.md [hunk 2]
```diff
@@ -177,6 +177,33 @@ Statement summary tables 现在还存在以下限制：
...
+
+> **注意：**
+>
+> - 当开启持久化后，由于不再于内存中维护历史数据，因此[参数配置](#参数配置) 一节所描述的 `tidb_stmt_summary_history_size` 将不再生效，而是由 [`tidb_stmt_summary_file_max_days`](/tidb-configuration-file.md#tidb_stmt_summary_file_max_days-从-v660-版本开始引入)、 [`tidb_stmt_summary_file_max_size`](/tidb-configuration-file.md#tidb_stmt_summary_file_max_size-从-v660-版本开始引入) 和 [`tidb_stmt_summary_file_max_backups`](/tidb-configuration-file.md#tidb_stmt_summary_file_max_backups-从-v660-版本开始引入) 这三项配置来决定历史数据在磁盘上的保留数量和时间。
```
Comment (Oreoxmt) [inline statement-summary-tables.md:204]:
```suggestion
> - 当开启持久化后，由于不再于内存中维护历史数据，因此[参数配置](#参数配置)一节所描述的 `tidb_stmt_summary_history_size` 将不再生效，而是由 [`tidb_stmt_summary_file_max_days`](/tidb-configuration-file.md#tidb_stmt_summary_file_max_days-从-v660-版本开始引入)、[`tidb_stmt_summary_file_max_size`](/tidb-configuration-file.md#tidb_stmt_summary_file_max_size-从-v660-版本开始引入) 和 [`tidb_stmt_summary_file_max_backups`](/tidb-configuration-file.md#tidb_stmt_summary_file_max_backups-从-v660-版本开始引入) 这三项配置来决定历史数据在磁盘上的保留数量和时间。
```

File: statement-summary-tables.md [hunk 3]
```diff
@@ -177,6 +177,33 @@ Statement summary tables 现在还存在以下限制：
 
 - TiDB server 重启后以上 4 张表的 statement summary 会全部丢失。因为 statement summary tables 全部都是内存表，不会持久化数据，所以一旦 server 被重启，statement summary 随之丢失。
 
+## 持久化 statements summary
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline statement-summary-tables.md:184]:
```suggestion
> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。语法和实现可能会在 GA 前发生变化。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```
https://github.com/pingcap/docs-cn/pull/12177

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
 + 该值作为系统变量 [`tidb_enable_ddl`](/system-variables.md#tidb_enable_ddl) 的初始值。
 + 在 v6.3.0 之前，该功能由配置项 `run-ddl` 进行设置。
 
+### `tidb_stmt_summary_enable_persistent` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> Statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:815]:
```suggestion
> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。语法和实现可能会在 GA 前发生变化。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
+
++ 用于控制是否开启 statements summary 持久化。
++ 默认值：false
++ 详情参考[持久化 statements summary](/statement-summary-tables.md#持久化-statements-summary)。
+
+### `tidb_stmt_summary_filename` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:825]:
```suggestion
> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。语法和实现可能会在 GA 前发生变化。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
+
++ 当开启了 statements summary 持久化时，该配置用于指定持久化数据所写入的文件名称。
++ 默认值：tidb-statements.log
+
+### `tidb_stmt_summary_file_max_days` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:834]:
```suggestion
> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。语法和实现可能会在 GA 前发生变化。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: tidb-configuration-file.md [hunk 7]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
+
++ 当开启了 statements summary 持久化时，该配置用于指定持久化数据文件所保留的最大天数。
++ 默认值：3
++ 可结合数据保留时长需求与磁盘空间占用适当调整。
+
+### `tidb_stmt_summary_file_max_size` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:844]:
```suggestion
> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。语法和实现可能会在 GA 前发生变化。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: tidb-configuration-file.md [hunk 8]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
+
++ 当开启了 statements summary 持久化时，该配置用于限制持久化数据单个文件最大尺寸 (MiB)。
++ 默认值：64
++ 可结合数据保留时长需求与磁盘空间占用适当调整。
+
+### `tidb_stmt_summary_file_max_backups` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:854]:
```suggestion
> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。语法和实现可能会在 GA 前发生变化。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: statement-summary-tables.md [hunk 9]
```diff
@@ -177,6 +177,33 @@ Statement summary tables 现在还存在以下限制：
...
+
+如[目前的限制](#目前的限制)一节所描述，默认情况下 statements summary 只在内存中维护，一旦 TiDB server 发生重启，所有 statements summary 数据都会丢失。自 v6.6.0 版本起，TiDB 实验性地提供了配置项 [`tidb_stmt_summary_enable_persistent`](/tidb-configuration-file.md#tidb_stmt_summary_enable_persistent-从-v660-版本开始引入) 来允许用户控制是否开启 statements summary 持久化。
```
Comment (Oreoxmt) [inline statement-summary-tables.md:186]:
```suggestion
如[目前的限制](#目前的限制)一节所描述，默认情况下 statements summary 只在内存中维护，一旦 TiDB server 发生重启，所有 statements summary 数据都会丢失。自 v6.6.0 起，TiDB 提供了 [`tidb_stmt_summary_enable_persistent`](/tidb-configuration-file.md#tidb_stmt_summary_enable_persistent-从-v660-版本开始引入) 配置项（实验特性）来允许用户控制是否开启 statements summary 持久化。
```

File: tidb-configuration-file.md [hunk 10]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
++ 默认值：false
++ 详情参考[持久化 statements summary](/statement-summary-tables.md#持久化-statements-summary)。
+
+### `tidb_stmt_summary_filename` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
+
++ 当开启了 statements summary 持久化时，该配置用于指定持久化数据所写入的文件名称。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:827]:
```suggestion
+ 当开启了 statements summary 持久化时，该配置用于指定持久化数据所写入的文件。
```

File: tidb-configuration-file.md [hunk 11]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
++ 默认值：tidb-statements.log
+
+### `tidb_stmt_summary_file_max_days` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
+
++ 当开启了 statements summary 持久化时，该配置用于指定持久化数据文件所保留的最大天数。
++ 默认值：3
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:837]:
```suggestion
+ 默认值：3
+ 单位：天
```

File: tidb-configuration-file.md [hunk 12]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
++ 详情参考[持久化 statements summary](/statement-summary-tables.md#持久化-statements-summary)。
+
+### `tidb_stmt_summary_filename` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
+
++ 当开启了 statements summary 持久化时，该配置用于指定持久化数据所写入的文件名称。
++ 默认值：tidb-statements.log
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:828]:
```suggestion
+ 默认值："tidb-statements.log"
```

File: tidb-configuration-file.md [hunk 13]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
++ 默认值：3
++ 可结合数据保留时长需求与磁盘空间占用适当调整。
+
+### `tidb_stmt_summary_file_max_size` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> statements summary 持久化目前为实验性特性，不建议在生产环境中使用。
+
++ 当开启了 statements summary 持久化时，该配置用于限制持久化数据单个文件最大尺寸 (MiB)。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:846]:
```suggestion
+ 当开启了 statements summary 持久化时，该配置用于限制持久化数据单个文件的大小 (MiB)。
```

File: tidb-configuration-file.md [hunk 14]
```diff
@@ -808,6 +808,55 @@ TiDB 服务状态相关配置。
...
++ 默认值：64
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:847]:
```suggestion
+ 默认值：64
+ 单位：MiB
```

File: statement-summary-tables.md (hunk 9 above)
Comment (shichun-0415) [reply statement-summary-tables.md:186]: statements summary 持久化是实验特性，其控制开关参数感觉不能叫实验特性。这里似乎不强调也没问题，而且开关本身不存在实验和非实验的区别，只是实验特性一般开关是默认关闭的。

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: statement-summary-tables.md [hunk 15]
```diff
@@ -177,6 +177,33 @@ Statement summary tables 现在还存在以下限制：
 
 - TiDB server 重启后以上 4 张表的 statement summary 会全部丢失。因为 statement summary tables 全部都是内存表，不会持久化数据，所以一旦 server 被重启，statement summary 随之丢失。
 
+## 持久化 statements summary
+
+> **警告：**
+>
+> statements summary 持久化目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+如[目前的限制](#目前的限制)一节所描述，默认情况下 statements summary 只在内存中维护，一旦 TiDB server 发生重启，所有 statements summary 数据都会丢失。自 v6.6.0 版本起，TiDB 实验性地提供了配置项 [`tidb_stmt_summary_enable_persistent`](/tidb-configuration-file.md#tidb_stmt_summary_enable_persistent-从-v660-版本开始引入) 来允许用户控制是否开启 statements summary 持久化。
```
Comment (Oreoxmt) [inline statement-summary-tables.md:186]: 6.6 支持持久化 statements summary，可以删除**目前的限制**一节并在前面的示例中添加开启 statements summary 持久化的说明。如果需要 GA 之后再删除，也可以先修改目前的限制的描述，添加持久化 statements summary 实验特性功能的链接

Comment (Oreoxmt) [inline statement-summary-tables.md:186]:
```suggestion
如[目前的限制](#目前的限制)一节所描述，默认情况下 statements summary 只在内存中维护，一旦 TiDB server 发生重启，所有 statements summary 数据都会丢失。自 v6.6.0 起，TiDB 实验性地提供了配置项 [`tidb_stmt_summary_enable_persistent`](/tidb-configuration-file.md#tidb_stmt_summary_enable_persistent-从-v660-版本开始引入) 来允许用户控制是否开启 statements summary 持久化。
```

File: statement-summary-tables.md [hunk 16]
```diff
@@ -177,6 +177,33 @@ Statement summary tables 现在还存在以下限制：
...
+
+如果要开启 statements summary 持久化，可以在 TiDB 配置文件中添加如下配置：
+
+```toml
+[instance]
+tidb_stmt_summary_enable_persistent = true
+# 以下配置为默认值，可根据需求调整。
+# tidb_stmt_summary_filename = "tidb-statements.log"
+# tidb_stmt_summary_file_max_days = 3
+# tidb_stmt_summary_file_max_size = 64 # MB
```
Comment (Oreoxmt) [inline statement-summary-tables.md:196]:
```suggestion
# tidb_stmt_summary_file_max_size = 64 # MiB
```

File: statement-summary-tables.md (hunk 15 above)
Comment (mornyx) [reply statement-summary-tables.md]: 期望是 GA 后再删除，PTAL: https://github.com/pingcap/docs-cn/pull/12677/commits/928b17e7efd35f6cd18aaeb4c0f1d282be12c141

File: statement-summary-tables.md [hunk 17]
```diff
@@ -177,6 +177,35 @@ Statement summary tables 现在还存在以下限制：
 
 - TiDB server 重启后以上 4 张表的 statement summary 会全部丢失。因为 statement summary tables 全部都是内存表，不会持久化数据，所以一旦 server 被重启，statement summary 随之丢失。
 
+自 v6.6.0 起，TiDB 为 statements summary 实验性地引入了持久化能力，该功能默认未开启，详情请参考[持久化 statements summary](#持久化-statements-summary)。
```
Comment (shichun-0415) [inline statement-summary-tables.md:180]: Line 176-180，建议修改为：
> 由于 statement summary tables 都存储在内存内，TiDB server 重启后，statement summary 会全部丢失。为解决该问题，TiDB v6.6.0 实验性地引入了 statement summary 持久化功能，开启该功能后，历史数据不再存储在内存内，而是直接写入磁盘，TiDB server 重启后，数据也依然可用。

Comment (shichun-0415) [reply statement-summary-tables.md:180]: @mornyx @Oreoxmt PTAL

Comment (shichun-0415) [reply statement-summary-tables.md]: Done

File: statement-summary-tables.md [hunk 18]
```diff
@@ -173,9 +173,36 @@ select * from information_schema.statements_summary_evicted;
 
 ## 目前的限制
 
-Statement summary tables 现在还存在以下限制：
+由于 statement summary tables 默认都存储在内存中，TiDB server 重启后，statement summary 会全部丢失。
 
-- TiDB server 重启后以上 4 张表的 statement summary 会全部丢失。因为 statement summary tables 全部都是内存表，不会持久化数据，所以一旦 server 被重启，statement summary 随之丢失。
+为解决该问题，TiDB v6.6.0 实验性地引入了 [statement summary 持久化](#持久化-statements-summary)功能，该功能默认未关闭。开启该功能后，历史数据不再存储在内存内，而是直接写入磁盘，TiDB server 重启后，历史数据也依然可用。
```
Comment (Oreoxmt) [inline statement-summary-tables.md:178]:
```suggestion
为解决该问题，TiDB v6.6.0 实验性地引入了 [statement summary 持久化](#持久化-statements-summary)功能，该功能默认为关闭。开启该功能后，历史数据不再存储在内存内，而是直接写入磁盘，TiDB server 重启后，历史数据也依然可用。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**